packit_service/models.py

```diff
@@ -3328,6 +3357,39 @@ def get_by_id(cls, group_id: int) -> Optional["TFTTestRunGroupModel"]:
         with sa_session_transaction() as session:
             return session.query(TFTTestRunGroupModel).filter_by(id=group_id).first()
 
+    @classmethod
+    def get_running(cls, commit_sha: str) -> Iterable[str]:
+        """Get list of currently running Testing Farm runs matching the passed
+        arguments.
+
+        Args:
+            commit_sha: Commit hash that is used for filtering the running jobs.
+
+        Returns:
+            An iterator over (string) request IDs that represent matching TF
+            runs that are _new_ (to be triggered), _queued_ (already submitted
+            to the TF), or _running_.
+        """
+        q = (
+            select(TFTTestRunTargetModel.pipeline_id)
+            .join(TFTTestRunGroupModel)
+            .join(PipelineModel)
+            .join(ProjectEventModel)
+            .filter(
+                ProjectEventModel.commit_sha == commit_sha,
+                TFTTestRunTargetModel.status.in_(
+                    (
+                        TestingFarmResult.new,
```

Comment:
the `new` ones are not yet submitted IIRC (=> no pipeline_id), is that ok?

Reply:
I’ll check

Reply:
Ya, you're right… At the same time it complicates handling of this state a bit…

I admit that I didn't think of updating the statuses on the database side when writing this, but
1. babysitting should be definitely able to pick up on this
2. I suppose that Copr announces canceled builds on Fedora Messaging
3. I expect TF to send a request when the “cancel” action goes through
4. if we want to handle the DB state during the cancelling, which we should, then the listing of just IDs for cancelling will not be enough :/ I will need to adjust that